xlconnect.Rmd

```diff
@@ -0,0 +1,544 @@
+setCellStyle(wbcopy, 
+             formula = "Input!$B$2:$B$2", 
+             cellstyle = cstitle1)
+setCellStyle(wbcopy, 
+             formula = "Input!$B$3:$B$3", 
+             cellstyle = cstitle2)
+setCellStyle(wbcopy, 
+             formula = "Input!$C$5:$E$425", 
+             cellstyle = csinp)
+setCellStyle(wbcopy, 
+             formula = "Input!$B$5:$B$425", 
+             cellstyle = cscalc)
+if (!existsCellStyle(wbcopy, "csDate")) {
+  csDate <- createCellStyle(wbcopy, name = "DateFormat")
+  setDataFormat(csDate, format = "yyyy-mm-dd")
+}
+setCellStyle(wbcopy, sheet = "Input", 
+             row = 5:425, 
+             col = 2, 
+             cellstyle = csDate)
+
+# Save workbook - this actually writes the file
+saveWorkbook(wbcopy)
+```
+
+## Add a graph
+
+First, create an image out of a graph
+```{r , echo = TRUE, eval = FALSE}
+# If previous chunks have not been run, the file may not exist
+if(!exists("xlconnect-demo/XLConnectDemo.xlsx")) {
+    # Write and save a data frame into a specific named region of a workbook
+  writeNamedRegionToFile("xlconnect-demo/XLConnectDemo.xlsx", 
+                       data = swissfranc, name = "SwissFrancName", 
+                       formula = "NamedRegionEx!$B$2")
+} 
+# Load the demo file
+wb <- loadWorkbook("xlconnect-demo/XLConnectDemo.xlsx", create = TRUE)
+
+# Prepare object
+currencies <- names(swissfranc)[-1]
+gcurr <- reshape(swissfranc, 
+                 varying = currencies, 
+                 direction = "long", 
+                 v.names = "Value", 
+                 times = currencies, 
+                 timevar = "Currency")
+
+# Use ggplot2 to create a graph
+require(ggplot2)
+p <- ggplot(gcurr, aes(Date, Value, colour = Currency)) +
+  geom_line() + stat_smooth(method = "loess") +
+  scale_y_continuous("Exchange Rate CHF/Curr") +
+  labs(title = paste0("CHF vs ", paste(currencies, collapse = ", ")), x = "") + 
+  theme(axis.title.y = element_text(size = 10, angle = 90, vjust = 0.3))
+print(p)
+
+# Save plot to as a png
+dev.copy(png, "xlconnect-demo/swissfrancplot.png")
+dev.off()
+```
+
+Then, add the image to the output file
+```{r , echo = TRUE, eval = FALSE}
+# Define the location in the output file
+createName(wb, name = "graph", formula = "NamedRegionEx!$H$2")
+
+# Add image to the file
+addImage(wb, 
+         filename = "xlconnect-demo/swissfrancplot.png", 
+         name = "graph", 
+         originalSize = TRUE)
+
+# Save workbook - this actually writes the file
+saveWorkbook(wb)
+```
+
+
+## Other functionalities 
+
+Let's perform some usual tasks on a template and use few more interesting XLConnect functionalities like:
+
+- ensure values get updated when opening a workbook
+
+- check if a sheet is visible
+
+- check if a named region exists
+
+- hide or unhide sheets
+
+- append a named region
+
+- get the coordinates of a named region in a sheet
+
+- additional formatting, e.g. setting the column width
+
+- remove sheets and named region
```

Comment:
```suggestion
- remove sheets and named regions
```
don't mind if these bullet points are formulated in singular or plural, but would use the same in one line and it would otherwise miss an "a"